datafusion/expr/src/udf.rs

```diff
@@ -481,6 +537,7 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
     /// This function must consistently return the same type for the same
     /// logical input even if the input is simplified (e.g. it must return the same
     /// value for `('foo' | 'bar')` as it does for ('foobar').
+    #[deprecated(since = "45.0.0", note = "Use `return_type_from_args` instead")]
```

Comment:
Can you also please update the comments on `return_type` to refer to `return_type_from_args` (it currently talks about `return_type_from_exprs`:

```rust
    /// If you provide an implementation for [`Self::return_type_from_exprs`],
    /// DataFusion will not call `return_type` (this function). In this case it
    /// is recommended to return [`DataFusionError::Internal`].
```